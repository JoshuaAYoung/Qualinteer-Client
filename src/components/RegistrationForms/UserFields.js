
import React, { Component } from 'react';
import ValidationError from '../../utilities/ValidationError';

export default class UserFields extends Component {
  render() {
    const { emailError, passwordError1, passwordError2, passwordError3, passwordError4 } = this.props.errorProps
    return (
      <>
        <div>
          <label
            htmlFor='email'
            className='registerLabel'>
            Email Address:
          </label>
          <br />
          <input
            type='text'
            name='email'
            id='email'
            className='registerInput'
            onChange={ev => this.props.addTempValidation('email', ev.target.value)}
            required
          />
        </div>
        {emailError && <ValidationError message={emailError} />}
        <br />
        <div>
          <label
            htmlFor='password'
            className='registerLabel'>
            Password:
          </label>
          <br />
          <input
            type='password'
            name='password'
            id='password'
            className='registerInput'
            onChange={ev => this.props.addTempValidation('password', ev.target.value)}
            required
          />
        </div>
        {passwordError1 && <ValidationError message={passwordError1} />}
        {passwordError2 && <ValidationError message={passwordError2} />}
        {passwordError3 && <ValidationError message={passwordError3} />}
        {passwordError4 && <ValidationError message={passwordError4} />}
        <br />
        <div>
          <label
            htmlFor='full_Name'
            className='registerLabel'>
            Full Name:
          </label>
          <br />
          <input
            type='text'
            name='full_Name'
            id='full_Name'
            className='registerInput'
            onChange={ev => this.props.addTempValue('full_name', ev.target.value)}
            required
          />
        </div>
        <br />
      </>
    )
  }
}