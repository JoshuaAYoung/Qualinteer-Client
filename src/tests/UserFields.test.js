import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UserFields from '../components/RegistrationForms/UserFields';

describe('UserFields Component', () => {
  const props = {
    errorProps: {
      emailError: 'test error 1',
      passwordError1: 'test error 1',
      passwordError2: 'test error 1',
      passwordError3: 'test error 1',
      passwordError4: 'test error 1',
    },
  };

  it('UserFields Smoke Test', () => {
    const div = document.createElement('div');

    ReactDOM.render(<UserFields {...props} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('UserFields Snapshot Test', () => {
    const wrapper = shallow(<UserFields {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
