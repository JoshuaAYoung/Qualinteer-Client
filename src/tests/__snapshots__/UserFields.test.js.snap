// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserFields Component UserFields Snapshot Test 1`] = `
<Fragment>
  <div
    className="inputContainer"
  >
    <input
      className="registrationInput"
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="Your Email"
      required={true}
      type="text"
    />
  </div>
  <ValidationError
    message="test error 1"
  />
  <div
    className="inputContainer"
  >
    <input
      className="registrationInput passwordInput"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
    />
    <button
      className="showRegistrationButton"
      onClick={[Function]}
      type="button"
    />
  </div>
  <ValidationError
    message="test error 1"
  />
  <ValidationError
    message="test error 1"
  />
  <ValidationError
    message="test error 1"
  />
  <ValidationError
    message="test error 1"
  />
  <div
    className="inputContainer"
  >
    <input
      className="registrationInput"
      id="full_Name"
      name="full_Name"
      onChange={[Function]}
      placeholder="Your Full Name"
      required={true}
      type="text"
    />
  </div>
</Fragment>
`;
